packages/cli/generation/ir-generator/src/converters/services/convertHttpService.ts

```diff
@@ -380,10 +386,25 @@ function convertTransportToEncoding(transport: Transport): Encoding {
             };
         case "grpc":
             return {
-                json: undefined,
+                json: serviceHasHttpEndpoints(service) ? {} : undefined,
```

Comment:
Discussed offline - we can nix this for now. Eventually we'll want to support per-endpoint level encoding, but for now we'll keep the contract as:

1. By default, all endpoints inherit the transport/encoding specified on the service.
2. If the endpoint defines its own transport/encoding, it will override the service-level value.